### PR TITLE
#2709 - shows notification bar with error message when email activation link is wrong

### PR DESCRIPTION
**Pull request by Moejoe90**
Also refactors the toast component to use `cy.get` assertions

### CODE REVIEW

File: app/test-e2e/link-triggered-flows/WalletVerify.spec.e2e.ts [hunk 1]
```diff
@@ -62,6 +62,12 @@ describe("Verify Wallet", () => {
 
       cy.get(tid("light-wallet-login-with-email-email-field")).contains(email);
 
+      // Asserts if error toast shows up
+      // @SEE https://github.com/Neufund/platform-frontend/issues/2709
+      cy.get(tid("modules.auth.sagas.verify-user-email.toast.verification-failed"));
```
Comment (desfero) [inline app/test-e2e/link-triggered-flows/WalletVerify.spec.e2e.ts]: can we use additionally `should('exist`)` as it's more appropriate and easy to figure out that it should exist

File: app/test-e2e/settings/ChangeEmail.spec.e2e.ts [hunk 2]
```diff
@@ -67,8 +66,8 @@ describe("Change Email", () => {
 
     acceptWallet();
 
-    // assert if error message has pooped in
-    assertNotificationExists("profile-email-exists");
+    // assert if error message has popped in
+    cy.get("profile-email-exists");
```
Comment (desfero) [inline app/test-e2e/settings/ChangeEmail.spec.e2e.ts]: same here and on line 80 and 106. And please use `tid`

File: app/test-e2e/utils/assertions.ts [hunk 3]
```diff
@@ -135,12 +135,13 @@ export const assertEmailChangeFlow = (): void => {
...
 };
 
+// @deprecated us cy.get()directly
 export const assertNotificationExists = (selector: string): void => {
   cy.get(notificationTid(selector)).should("exist");
 };
 
 export const assertEmailPendingChange = (email: string, newEmail: string): void => {
-  assertNotificationExists("profile-email-change-success");
+  cy.get("profile-email-change-success");
```
Comment (desfero) [inline app/test-e2e/utils/assertions.ts]: `should('exist')`

File: app/components/shared/Toast.tsx [hunk 4]
```diff
@@ -8,15 +8,20 @@ import "react-toastify/dist/ReactToastify.css";
 
 const ToastContainer = () => <ToastifyContainer autoClose={TOAST_COMPONENT_DELAY} />;
 
-const getOptions = (options?: TDataTestId) =>
-  IS_CYPRESS && options ? { className: options["data-test-id"] } : undefined;
+const ComponentWithTestId = (children: TTranslatedString, options?: TDataTestId) =>
```
Comment (desfero) [inline app/components/shared/Toast.tsx]: do you think we can refactor this component to:

```
getComponentWIthTestId = (children: ..., dataTestId?: string) => ...
```

File: app/test-e2e/utils/assertions.ts (hunk 3 above)
Comment (Moejoe90) [reply app/test-e2e/utils/assertions.ts]: :+1:

**Comment by Moejoe90** [comment]
This couldn't be tested with commit deployment testing because you need to use a link

**Pull request merged**